example/tpm_ibmtss2/time_decrypt.c

```diff
@@ -0,0 +1,408 @@
+        goto err;
+
+    fprintf(stderr, "fclose()\n");
+    if (fclose(fp) != 0)
+        goto err;
+    fp = NULL;
+
+    fprintf(stderr, "RsaPemToTPM()\n");
+    memset(&le_in, 0, sizeof(le_in));
+    if (RsaPemToTPM(pkey, &le_in.inPrivate, &le_in.inPublic,
+                    TPM_ALG_NULL, TPM_ALG_SHA256) < 0) {
+        fprintf(stderr, "Could not convert key to public\n");
+        goto err;
+    }
+    le_in.hierarchy = TPM_RH_NULL;
+
+    if (ciphertext_len > le_in.inPublic.publicArea.unique.rsa.t.size) {
+        fprintf(stderr, "The -n parameter is larger then the size of modulus: %d > %d\n",
+                ciphertext_len, le_in.inPublic.publicArea.unique.rsa.t.size);
+        goto err;
+    }
+
+    if ((rc = TSS_Create(&tssContext)) != 0) {
+        fprintf(stderr, "Could not create TSS context: 0x%x\n", rc);
+        goto err;
+    }
+
+    if ((rc = TSS_Execute(tssContext,
+                          (RESPONSE_PARAMETERS *)&le_out,
+                          (COMMAND_PARAMETERS *)&le_in,
+                          NULL,
+                          TPM_CC_LoadExternal,
+                          TPM_RH_NULL, NULL, 0,
+                          TPM_RH_NULL, NULL, 0,
+                          TPM_RH_NULL, NULL, 0,
+                          TPM_RH_NULL, NULL, 0)) != 0) {
+        fprintf(stderr, "Could not execute LoadExternal command: 0x%x\n", rc);
+        goto err;
+    }
+
+    fprintf(stderr, "Decrypting ciphertexts using TPM 2 ...\n");
+
+    int c = 0, dec_fail = 0, msg_fail = 0;
+
+    while ((r_ret = read(in_fd, ciphertext, ciphertext_len)) > 0) {
+        if (r_ret != ciphertext_len) {
+            fprintf(stderr, "read less data than expected (truncated file?)\n");
+            goto err;
+        }
+        c++;
+
+        memset(&rd_in, 0, sizeof(rd_in));
+        rd_in.keyHandle = le_out.objectHandle;
+        rd_in.cipherText.t.size = ciphertext_len;
+        memcpy(rd_in.cipherText.t.buffer, ciphertext, ciphertext_len);
+        rd_in.inScheme.scheme = TPM_ALG_RSAES;
+        rd_in.label.t.size = 0;
+
+        time_before = get_time_before();
+
+        r_ret = TSS_Execute(tssContext,
+                            (RESPONSE_PARAMETERS *)&rd_out,
+                            (COMMAND_PARAMETERS *)&rd_in,
+                            NULL,
+                            TPM_CC_RSA_Decrypt,
+                            TPM_RS_PW, "", 0,
+                            TPM_RH_NULL, NULL, 0,
+                            TPM_RH_NULL, NULL, 0,
+                            TPM_RH_NULL, NULL, 0);
+
+        time_after = get_time_after();
+
+        if (r_ret != 0) {
+            if (verbose) {
+                fprintf(stderr, "Decryption failure: 0x%x\n", r_ret);
+                fprintf(stderr, "%x\n", ciphertext[0]);
+            }
+            dec_fail++;
+        } else if (rd_out.message.t.size != 48) {
+            /* Newer OpenSSL will not fail on decyrption errors but return
+             * longer messages.
+             */
```

Comment:
same as in the readme, with implicit rejection there should be no unexpected sizes if the test vectors are correct

Reply:
Correct, There are no more unexpected sizes. 

I thought the fact that no more unexpected sizes are returned now was due to me switching libtpms to use the Carmichael function for keys >= 2048 bit but this doesn't seem to be the case (anymore? due to update of openssl on my system?). Now none of the tests that ran returned unexpected sizes anymore.